Makefile

```diff
@@ -1,5 +1,5 @@
 CXX := g++
-CXXFLAGS := -std=c++2a -O3 -Wall -Wpedantic -MMD
+CXXFLAGS := -std=c++17 -O3 -Wall -Wpedantic -MMD

```

Comment:
obv discussion we had about this before, but i'll enumerate here so it's in writing on the repo:

C++17 makes some stuff not possible rn, i.e. we can't do struct initialization like `{.x = 5}`. Easy fix, we can just use constructors until this gets added in C++20. Might be useful to make a quick PR to get that & we can merge that before this PR just so we don't get warnings when we merge this. Happy to take care of that myself